File: pgm/ABC_Java/ABC_258_A.java
import java.io.*;
// import java.math.*;
import java.time.*;
import java.time.format.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final int K = Integer.parseInt (reader.readLine());
        final String HHmm = "HH:mm";
        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern (HHmm);
        final LocalDateTime datetime = LocalDateTime.of (2022, 7, 2, 21, 0, 0);
        System.out.println (datetime.plusMinutes (K).format (formatter));
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_258_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
import java.time.LocalTime
import java.time.format.DateTimeFormatter
// import java.util.Scanner
// import java.util.StringTokenizer

import kotlin.math.*

const val HHmm: String = "HH:mm"
const val hh: Int = 21
const val mm: Int = 0

fun main() {
    val K: Long = readln().toLong()
    println(DateTimeFormatter
        .ofPattern(HHmm)
        .format(LocalTime.of(hh, mm)
        .plusMinutes(K)))
    kotlin.system.exitProcess(0)
}